rs/tests/consensus/orchestrator/node_registration_test.rs

```diff
@@ -1,10 +1,138 @@
-#[rustfmt::skip]
+/* tag::catalog[]
 
-use anyhow::Result;
+Goal:: Test the node registration process by mocking the HSM signing.
+
+Runbook::
+. Setup an IC with 1-node NNS and 1 unassigned node with a node provider and a node operator principals (we use the same one).
+. Remove the node from the registry.
+. Wait for the registry update and make sure we have no unassigned nodes.
+. Delete crypto key on the unassigned node and restart the replica and the csp process again.
+. Restart the replica process.
+. Wait for the registry update and make sure we have 1 unassigned nodes.
+
+Success:: We end the test again with 1 registered unassigned nodes.
 
+end::catalog[] */
+
+use anyhow::Result;
+use ic_base_types::PrincipalId;
+use ic_consensus_system_test_utils::ssh_access::execute_bash_command;
+use ic_nns_constants::GOVERNANCE_CANISTER_ID;
+use ic_nns_governance::pb::v1::NnsFunction;
+use ic_registry_subnet_type::SubnetType;
 use ic_system_test_driver::driver::group::SystemTestGroup;
+use ic_system_test_driver::driver::{ic::InternetComputer, test_env::TestEnv, test_env_api::*};
+use ic_system_test_driver::nns::{
+    submit_external_proposal_with_test_id, vote_execute_proposal_assert_executed,
+};
 use ic_system_test_driver::systest;
-use ic_tests::orchestrator::node_registration_test::{config, test};
+use ic_system_test_driver::util::{block_on, runtime_from_url};
+use registry_canister::mutations::node_management::do_remove_nodes::RemoveNodesPayload;
+use slog::info;
+use std::str::FromStr;
+
+fn config(env: TestEnv) {
```

Comment:
What about calling this `setup`? The `config` naming is deprecated.
```suggestion
fn setup(env: TestEnv) {
```

Reply:
Done, thanks